pytradfri/smart_task.py

```diff
@@ -0,0 +1,242 @@
+"""Smart tasks set timers to turn on/off lights in various ways.
+
+v1: Added support to show (not modify) states for wake up smart task.
+
+"""
+
+from datetime import datetime
+from .const import (
+    ATTR_CREATED_AT,
+    ATTR_ID,
+    ATTR_LIGHT_DIMMER,
+    ATTR_LIGHT_STATE,
+    ATTR_SMART_TASK_LIGHTS_OFF,
+    ATTR_SMART_TASK_NOT_AT_HOME,
+    ATTR_SMART_TASK_TRIGGER_TIME_INTERVAL,
+    ATTR_SMART_TASK_TRIGGER_TIME_START_HOUR,
+    ATTR_SMART_TASK_TRIGGER_TIME_START_MIN,
+    ATTR_SMART_TASK_TYPE,
+    ATTR_SMART_TASK_WAKE_UP,
+    ATTR_TRANSITION_TIME,
+    ATTR_REPEAT_DAYS,
+    ATTR_START_ACTION,
+    ROOT_START_ACTION,
+    ROOT_SMART_TASKS
+)
+
+
+class BitChoices(object):
+    """Helper class for bitwise dates.
+
+    http://stackoverflow.com/questions/3663898/representing-a-multi-select-field-for-weekdays-in-a-django-model
+    """
+
+    def __init__(self, choices):
+        """Initialize BitChoices class."""
+        self._choices = []
+        self._lookup = {}
+        for index, (key, val) in enumerate(choices):
+            index = 2**index
+            self._choices.append((index, val))
+            self._lookup[key] = index
+
+    def __iter__(self):
+        return iter(self._choices)
+
+    def __len__(self):
+        return len(self._choices)
+
+    def __getattr__(self, attr):
+        try:
+            return self._lookup[attr]
+        except KeyError:
+            raise AttributeError(attr)
+
+    def get_selected_keys(self, selection):
+        """Return a list of keys for the given selection."""
+        return [k for k, b in self._lookup.iteritems() if b & selection]
```

Comment:
Yes. `dict.items` works both in python 2 and 3, but is more efficient in python 3. `dict.iteritems` is more efficient in python 2 than `dict.items`.